lxd/patches.go

```diff
@@ -840,4 +841,73 @@ func patchStorageRenameCustomISOBlockVolumes(name string, d *Daemon) error {
 	return nil
 }
 
+// patchZfsSetContentTypeUserProperty adds the `lxd:content_type` user property to custom storage volumes. In case of recovery, this allows for proper detection of block-mode enabled volumes.
+func patchZfsSetContentTypeUserProperty(name string, d *Daemon) error {
+	s := d.State()
+
+	// Get all storage pool names.
+	pools, err := s.DB.Cluster.GetStoragePoolNames()
+	if err != nil {
+		return fmt.Errorf("Failed getting storage pool names: %w", err)
+	}
+
+	volTypeCustom := db.StoragePoolVolumeTypeCustom
+	customPoolVolumes := make(map[string][]*db.StorageVolume, 0)
+
+	err = s.DB.Cluster.Transaction(s.ShutdownCtx, func(ctx context.Context, tx *db.ClusterTx) error {
+		for _, pool := range pools {
+			// Get storage pool ID.
+			poolID, err := tx.GetStoragePoolID(ctx, pool)
+			if err != nil {
+				return fmt.Errorf("Failed getting storage pool ID of pool %q: %w", pool, err)
+			}
+
+			// Get the pool's custom storage volumes.
+			customVolumes, err := tx.GetStoragePoolVolumes(ctx, poolID, false, db.StorageVolumeFilter{Type: &volTypeCustom})
```

Comment:
The earlier commit looked like you were setting the content type user property into all block backed volumes.
But this patch seems to be only applying it to all custom volumes (even filesystem ones), but not instance volumes.

So there looks like a bit if an inconsistency here. I'm not sure which part is wrong though.

Reply:
I've changed it so that only custom volumes will have the `lxd:content_type` user property.